.gitlab-ci.yml

```diff
@@ -485,6 +485,31 @@ fedora upgrade:
       - new-installed-database.sql
       - new-upgraded-database.sql
 
+flawfinder: 
+  stage: test
+  needs: []
+  variables:
+    GIT_STRATEGY: fetch
+    GIT_SUBMODULE_STRATEGY: normal
+  script:
+    - yum install -y python3 python3-pip jq diffutils
+    - pip install flawfinder
+    - flawfinder --falsepositive --quiet --html . > flawfinder-all-vulnerabilities.html
+    - flawfinder --falsepositive --quiet --minlevel=5 --sarif . > flawfinder-output.json
+    # FlawFinder's --sarif output will display all vulnerabilities despite having --minlevel=5 specified.
+    # Therefore, we postprocess the results with jq and filter out findings where the vulnerability level is less than 5.
+    # Also in the SARIF output format, the vulnerabilities are ranked as 0.2/0.4/0.6/0.8/1.0 which correspond to the --minlevel=1/2/3/4/5 of FlawFinder.
+    # Additionally, we sort the results because individual findings are consistent across different runs, but their ordering may not be.
+    # Vulnerabilities can also be ignored in-line (/* Flawfinder: ignore */), but this option was chosen as to not clutter the codebase.
+    - jq 'del(.runs[] | .tool | .driver | .rules) | del(.runs[] | .results[] | select(.rank < 1)) | .runs[0].results|=sort_by(.fingerprints)' flawfinder-output.json > flawfinder-min-level5.json
+    # Diff against known vulnerabilities, but ignore the line number.
+    - diff -I 'startLine' flawfinder-min-level5.json flawfinder-ignorelist.json
```

Comment:
If we _ignore_ the line number in the diff, but leave it in the SARIF/.json report, that's going to add some seemingly-unnecessary noise/churn when the `flawfinder-ignorelist.json` changes.

At the risk of making that `jq` one-liner even more complicated…

It'd probably be best simply to _remove_ the `startLine`s from the SARIF/.json output:
```
jq 'del(.runs[] | .tool | .driver | .rules) | del(.runs[] | .results[] | select(.rank < 1)) | 
     del(.runs[] | .results[] | .locations[] | .region | .startLine) | 
     .runs[0].results|=sort_by(.fingerprints)' flawfinder-output.json > flawfinder-min-level5.json
```

Reply:
Good point @dlenski. I've removed the `startLine`s from the ignorelist, and added your suggested jq addition.
I've also moved the ignorelist to `tests/code_quality/flawfinder-ignorelist.json` to cut down on root directory clutter.

Reply:
Great, thanks! [`git diff a8b60af..7802c23`](https://github.com/MariaDB/server/compare/a8b60af..7802c23) looks good, should declutter future updates to the ignorelist a fair bit.